rusoto/core/src/request.rs

```diff
@@ -444,6 +444,10 @@ where
         message: format!("error building request: {}", err),
     })?;
 
+    // we use HTTP/2 with rustls, the host header is required to sign the request
+    // Google cloud storage rejects HTTP/2 payloads containing a host header
+    #[cfg(any(feature = "rustls", feature = "rustls-webpki"))]
+    hyper_headers.remove("host");
```

Comment:
Host header is required for HTTP/1.1 but not HTTP/2 https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2.3